src/modules/application/application.service.ts

```diff
@@ -60,8 +61,18 @@ export class ApplicationService {
    * return true if App with given namespace exists
    * @param namespace
    */
-  public async exists(namespace: string) {
-    return Boolean(await this.getByNamespace(namespace));
+  public async exists(
+    namespace: string,
+    parentOrg?: Organization,
+  ): Promise<boolean> {
+    const whereObjectClause: any = { namespace };
+    if (parentOrg) {
+      whereObjectClause.parentOrg = parentOrg;
+    }
+    const appExists = await this.applicationRepository.findOne({
+      where: whereObjectClause,
+    });
+    return Boolean(appExists);
```

Comment:
Why this change? Is it fixing a bug or just adding the ability to also have a check for the `parentOrg`. I would think that the `parentOrg` is implied in the namespace already. Like if a namespace like `finance.apps.ahmed.iam.ewc` is passed a `namespace`, aren't we already checking that the parent is `ahmed.iam.ewc` implicitly 🤔 ?

Reply:
While cleaning up duplicates, I discovered that, There are cases where we have duplicates with the same `namespace` and `parentOrg` so this is more like a check for if that namespace with that parentOrg already exists.

Reply:
Hmm, okay, I'm still not fully understanding when someone would use the `parentOrg` param. What I mean is that, if I'm querying for a role like `engineer.roles.ahmed.iam.ewc`, the parent org can only be `ahmed.iam.ewc`. So if someone were to make a query like `exists(engineer.roles.ahmed.iam.ewc, john.iam.ewc)`, that is obviously going to be false. Am I misunderstanding something here?

Reply:
Sorry for the misunderstanding. you're right. Dmitry clarified this to me this morning.